rsb-java/src/test/java/rsb/patterns/LocalServerStateTest.java

```diff
@@ -48,4 +55,35 @@ protected Participant createParticipant(
                 new Scope("/some/scope")).setConfig(config));
     }
 
+    /**
+     * Checks that deactivation is successful even after adding a method failed.
+     *
+     * @throws Exception
+     *             test error
+     */
+    @Test
+    public void deactivationAfterNewMethodFailure() throws Exception {
+        final LocalServer server = (LocalServer) createParticipant(
+                Utilities.createBrokenParticipantConfig());
+        server.activate();
+        try {
```

Comment:
Can we use this? https://junit.org/junit4/javadoc/4.12/org/junit/rules/ExpectedException.html

Reply:
This doesn't work here. `addMethod` and `deactivate` both can throw `RSBException`, but we only want to see it at the `addMethod` call. `deactivate` must still succeed. Using the exception annotation would make both indistinguishable and hence the test would also succeed in the unintended case that `deactivate` throws.

Reply:
I will comment this